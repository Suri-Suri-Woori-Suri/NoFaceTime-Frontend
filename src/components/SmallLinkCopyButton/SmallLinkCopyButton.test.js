import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import SmallLinkCopyButton from './SmallLinkCopyButton';

describe('<SmallLinkCopyButton/>', () => {
  afterEach(cleanup);

  const mockButtonName = 'properly render';
  const mockLink = 'https://www.twofacetim.xyz';
  const mockClickEventFunction = jest.fn();

  const wrapper = mount(
    <SmallLinkCopyButton
      buttonName={mockButtonName}
      link={mockLink}
      clickEventFunction={mockClickEventFunction} />
  );

  it('should be render button name correctly', () => {
    expect(wrapper.containsMatchingElement(mockButtonName)).toEqual(true);
  });

  it('should be invoked when button is clicked', () => {
    render(
      <SmallLinkCopyButton
        buttonName={mockButtonName}
        link={mockLink}
        clickEventFunction={mockClickEventFunction} />
    );

    fireEvent.click(screen.getByText(mockButtonName));
    expect(mockClickEventFunction).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
