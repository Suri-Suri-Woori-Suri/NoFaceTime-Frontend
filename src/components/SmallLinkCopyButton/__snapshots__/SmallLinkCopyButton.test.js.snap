// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallLinkCopyButton/> matches snapshot 1`] = `
<SmallButton
  buttonName="properly render"
  clickEventFunction={[MockFunction]}
  link="https://www.twofacetim.xyz"
>
  <button
    className="SmallLinkCopyButton"
    id="https://www.twofacetim.xyz"
    onClick={[Function]}
    value="https://www.twofacetim.xyz"
  >
    properly render
  </button>
</SmallButton>
`;
